processing/src/main/java/io/druid/query/filter/JavaScriptDimFilter.java

```diff
@@ -79,8 +101,45 @@ public DimFilter optimize()
   public String toString()
   {
     return "JavaScriptDimFilter{" +
-           "dimension='" + dimension + '\'' +
+           "dimensions=['" + Joiner.on("', '").join(dimensions) + "']" +
            ", function='" + function + '\'' +
+           ", rowBased='" + rowBased + '\'' +
            '}';
   }
-}
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof JavaScriptDimFilter)) {
+      return false;
+    }
+
+    JavaScriptDimFilter that = (JavaScriptDimFilter) o;
+
+    if (rowBased ^ that.rowBased) {
+      return false;
+    }
+    if (!function.equals(that.function)) {
+      return false;
+    }
+    return Arrays.equals(that.dimensions, dimensions);
+  }
+
+  public static JavaScriptDimFilter of(String dimension, String function) {
```

Comment:
minor nit, but can we move static functions to the top of the class? similar to other druid files